app/controllers/api/base_controller/renderer.rb

```diff
@@ -156,10 +172,16 @@ def collection_search(is_subcollection, type, klass)
 
         options = {:user => User.current_user}
         options[:order] = sort_options if sort_options.present?
-        options[:offset], options[:limit] = expand_paginate_params if paginate_params?
         options[:filter] = miq_expression if miq_expression
+        options[:offset] = params['offset'] if params['offset']
+        options[:limit] = params['limit'] if params['limit']
+
+        miq_expression.present? ? subquery_search(res, options) : [Rbac.filtered(res, options)]
+      end
 
-        Rbac.filtered(res, options)
+      def subquery_search(res, options)
```

Comment:
not clear offhand what subquery_search does (as called from 179 if miq_expression.present), might need a better name